web/packages/teleport/src/AccessRequests/utils.ts

```diff
@@ -22,55 +22,55 @@ import {
   isAfter,
 } from 'date-fns';
 
-interface TimeDuration {
+type TimeDuration = {
   timestamp: number;
   duration: Duration;
-}
-
-export function middleValues(start: Date, end: Date): TimeDuration[] {
-  const now = new Date();
+};
 
-  const roundDuration = (d: Date) =>
-    roundToNearestHour(
-      intervalToDuration({
-        start: now,
-        end: d,
-      })
-    );
+// Generate a list of middle values between start and end. The first value is the
+// session TTL that is rounded to the nearest hour. The rest of the values are
+// rounded to the nearest day. Example:
+//
+// created: 2021-09-01T00:00:00.000Z
+// start: 2021-09-01T01:00:00.000Z
+// end: 2021-09-03T00:00:00.000Z
+// now: 2021-09-01T00:00:00.000Z
+//
+// returns: [1h, 1d, 2d, 3d]
+export function middleValues(
+  created: Date,
+  start: Date,
+  end: Date
+): TimeDuration[] {
+  const getInterval = (d: Date) =>
+    intervalToDuration({
+      start: created,
+      end: d,
+    });
 
   const points: Date[] = [start];
 
-  if (isAfter(addDays(start, 1), end)) {
+  if (isAfter(addDays(created, 1), end)) {
```

Comment:
can we add a test for this case too? if i understand correctly, this will only produce one option and not sure i saw it on the test